examples/s3/v1beta1/directorybucket.yaml

```diff
@@ -0,0 +1,15 @@
+# https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-Endpoints.html
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: DirectoryBucket
+metadata:
+  annotations:
+    meta.upbound.io/example-id: s3/v1beta1/directorybucket
+    crossplane.io/external-name: ${Rand.RFC1123Subdomain}--usw2-az1--x-s3
```

Comment:
Expecting users to specify this format on every resource would be a pretty bad user experience, especially since there's no guarantee that the availability zone id in this name would match the location name in the `spec`.